src/blog/2023/06/Flowforge-as-a-No-Code-Ethernet_IP-to-Profinet-Protocol-Converter.md

```diff
@@ -0,0 +1,208 @@
+description: step-by-step guide for using flowforge as an industrial protocol converter
+date: 2023-06-19
+authors: ["richard-meyer"]
+image:
+tags:
+  - posts
+  - how-to
+---
+
+
+Frequently in industrial automation, there's a need for two devices that use different protocols to communicate with each other, requiring protocol conversion.  
+In this article, we present a mock scenario where flowforge is used to enable an Allen Bradley PLC, which uses ethernet/IP, to communicate with a Siemens PLC, which uses Profinet, using a no-code solution. This example is geared toward beginners and assumes that the end-user knows how to use PLCs, but may be using flowforge/node-red for the first time.
+
+<!--more-->
+
+# Premise
+
+![1](./images/ethip-to-profinet/e-to-p-1.png)
+
+Figure 1 above shows the layout of a mock production facility. Inside our facility, operations suggested adding stacklights as an extra visual aid for operators to get a quick status of its 4 conveyor lines, avoiding the need to constantly monitor the HMI/SCADA displays.  
+Engineering has suggested adding a siemens S7 1200 PLC with an IO link connection to 4 stacklights, with each line PLC sending basic status information to the stacklight PLC to control the stacklight outputs.  
+Line 1-3 PLCs are Siemens-based, and can communicate with the stacklight PLC natively over Profinet. But line 4 is an Allen Bradley PLC that uses ethernet/IP, and can't communicate with the stacklight PLC without some form of protocol conversion.  
+Traditionally, we'd use protocol gateway hardware, like anybus or red lion, to convert ethernet/IP to profinet.  
+But for this application, we will instead use flowforge, a pure software-based approach, to convert ethernet/IP to Profinet. Let's walk through the process.
+
+# Pre-Requisites and Set Up
+
+## Flowforge
+
+In addition to our two PLCs, we’ll need access to flowforge software. You can either self-host, or have flowforge manage your flowforge instance.
+
+In this example, we will be using a self-hosted flowforge instance running in a docker container.
+
+[https://github.com/flowforge/flowforge](https://github.com/flowforge/flowforge)
+
+## Data Treatment on Ethenet/IP PLC
+
+In our Allen Bradley line 4 PLC, we will send some arbitrary tags of various datatypes to the stacklight PLC for illustrative purposes, described in table 1 below -
+
+| **Tag** | **Data Type** | **Description** |
+| --- | --- | --- |
+| Conveyor\_RTS | BOOL | Conveyor Ready to Start |
+| Robot\_RTS | BOOL | Robot is Ready to Start |
+| Robot\_Position | REAL | Robot Arm position (degrees) |
+| Conveyor\_Running | BOOL | Conveyor is running |
+| Line4\_State | DINT | Line 4 Machine State |
+| Line4\_Fault | BOOL | Line 4 is faulted |
+
+Table 1 - Line 4 Tags to be sent to Stacklight PLC
+
+We can send any atomic data type we want, but it must be globally (controller) scoped.
+
+![2](./images/ethip-to-profinet/e-to-p-2.png)
```

Comment:
For each of the images, can we given them nice alt tags and descriptions?

E.g. `![description](./images/....png "alternative description")`